cfme/tests/infrastructure/test_providers.py

```diff
@@ -295,17 +315,77 @@ def test_delete_custom_attributes(self, rest_api, custom_attributes, from_detail
+            body.append({
+                'name': 'ca_name_{}'.format(uid),
+                'value': 'ca_value_{}'.format(uid),
+                'section': 'metadata'
+            })
+        if from_detail:
+            edited = []
+            for i in range(response_len):
+                edited.append(attributes[i].action.edit(**body[i]))
+        else:
+            for i in range(response_len):
+                body[i].update(attributes[i]._ref_repr())
+            edited = provider.custom_attributes.action.edit(*body)
+        assert len(edited) == response_len
+        for i in range(response_len):
+            assert edited[i].name == body[i]['name']
+            assert edited[i].value == body[i]['value']
+
+    @pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
+    @pytest.mark.tier(3)
+    @test_requirements.rest
+    @pytest.mark.parametrize(
+        'from_detail', [True, False],
+        ids=['from_detail', 'from_collection'])
+    def test_edit_custom_attributes_bad_section(self, rest_api, custom_attributes, from_detail):
+        """Test that editing custom attributes using REST API and adding invalid section fails.
+
+        Metadata:
+            test_flag: rest
+        """
+        attributes, provider = custom_attributes
+        response_len = len(attributes)
+        body = []
+        for _ in range(response_len):
+            body.append({'section': 'bad_section'})
         if from_detail:
-            changed_attrs = []
-            changed_attrs.append(custom_attributes[0].action.edit(**attrs0))
-            changed_attrs.append(custom_attributes[1].action.edit(**attrs1))
+            for i in range(response_len):
+                with error.expected('Api::BadRequestError'):
+                    attributes[i].action.edit(**body[i])
         else:
-            attrs0.update(custom_attributes[0]._ref_repr())
-            attrs1.update(custom_attributes[1]._ref_repr())
-            provider = rest_api.collections.providers[0]
-            changed_attrs = provider.custom_attributes.action.edit(attrs0, attrs1)
-        assert changed_attrs[0].name == "ca_new_name1" and changed_attrs[0].value == "ca_new_value1"
-        assert changed_attrs[1].name == "ca_new_name2" and changed_attrs[1].value == "ca_new_value2"
+            for i in range(response_len):
+                body[i].update(attributes[i]._ref_repr())
+            with error.expected('Api::BadRequestError'):
+                provider.custom_attributes.action.edit(*body)
+
+    @pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
+    @pytest.mark.tier(3)
+    @test_requirements.rest
+    def test_add_custom_attributes_bad_section(self, rest_api, setup_a_provider):
+        """Test that adding custom attributes with invalid section
+        to provider using REST API fails.
+
+        Metadata:
+            test_flag: rest
+        """
+        provider = rest_api.collections.providers[0]
```

Comment:
Couldn't we get this from current appliance? I think this is the preferred way now